src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -529,8 +528,22 @@ Status ServiceBasedNodeInfoAccessor::AsyncSubscribeToResources(
     const StatusCallback &done) {
   RAY_LOG(DEBUG) << "Subscribing node resources change.";
   RAY_CHECK(subscribe != nullptr);
-  auto status =
-      resource_sub_executor_.AsyncSubscribeAll(ClientID::Nil(), subscribe, done);
+
+  auto on_subscribe = [subscribe](const std::string &id, const std::string &data) {
```

Comment:
just change the callback of `AsyncSubscribeToResources` to use `NodeResourceChange`.

Reply:
fixed